lib/src/facts/aix/filesystem_resolver.cc

```diff
@@ -0,0 +1,114 @@
+                boost::split(tokens, line, boost::is_space(), boost::token_compress_on);
+                _filesystems.emplace(make_pair(stoi(tokens[1]), tokens[0]));
+                result.filesystems.emplace(move(tokens[0]));
+            }
+            return true;
+        });
+    }
+
+    void filesystem_resolver::collect_mountpoint_data(data& result)
+    {
+        for (const auto& mount : mountctl()) {
+            mountpoint m;
+            if (mount.vmt_gfstype == MNT_PROCFS) {
+                continue;
+            }
+            m.filesystem = _filesystems[mount.vmt_gfstype];
+            m.device = reinterpret_cast<char*>(vmt2dataptr(&mount, VMT_OBJECT));
+            m.name = reinterpret_cast<char*>(vmt2dataptr(&mount, VMT_STUB));
+
+            string opts = reinterpret_cast<char*>(vmt2dataptr(&mount, VMT_ARGS));
+            boost::split(m.options, opts, boost::is_any_of(","));
+
+            struct statfs64 fs;
+            if (0 != statfs64(const_cast<char*>(m.name.c_str()), &fs)) {
+                LOG_WARNING("Could not get fs data for {1}: {2}", m.name, errno);
+            } else {
+                m.size = fs.f_bsize*fs.f_blocks;
+                m.available = fs.f_bsize*fs.f_bfree;
+            }
+
+            _mounts[m.device] = m.name;
+            result.mountpoints.emplace_back(move(m));
+        }
+    }
+
+    void filesystem_resolver::collect_partition_data(data& result)
+    {
+        auto cu_dv = odm_class<CuDv>::open("CuDv");
+        auto cu_at = odm_class<CuAt>::open("CuAt");
+        for (auto& dv : cu_dv.query("PdDvLn=logical_volume/lvsubclass/lvtype")) {
+            partition p;
+            p.name = string("/dev/") + dv.name;
+
+            string query = (boost::format("name=%1%") % dv.name).str();
+            for (auto& at : cu_at.query(query)) {
+                if (0 == strcmp(at.attribute, "label")) {
+                    p.label = at.value;
+                } else if (0 == strcmp(at.attribute, "lvserial_id")) {
+                    struct lv_id id;
+                    // AAAAAAAABBBBBBBBCCCCCCCCDDDDDDDD.EEEE format
+                    // First four chunks are hexadecimal 32-bit integers.
+                    // After the dot is a decimal integer
+                    id.vg_id.word1 = stoi(string(at.value, 8), nullptr, 16);
+                    id.vg_id.word2 = stoi(string(at.value+8, 8), nullptr, 16);
+                    id.vg_id.word3 = stoi(string(at.value+16, 8), nullptr, 16);
+                    id.vg_id.word4 = stoi(string(at.value+24, 8), nullptr, 16);
+                    id.minor_num = stoi(string(at.value+33), nullptr, 10);
+
+                    struct querylv* lv;
+                    if (0 != lvm_querylv(&id, &lv, nullptr)) {
+                        LOG_WARNING("Could not get info for partition '{1}' from the LVM subsystem", p.name);
+                    } else if (!lv) {
+                        LOG_WARNING("querylv returned success but we got a null LV. WTF?");
```

Comment:
I'm gonna leave it. This is one of those "should never happen" situations. I'm only checking it because the documentation doesn't explicitly call it impossible.